zap/src/main/java/org/zaproxy/zap/spider/parser/SpiderHtmlParser.java

```diff
@@ -208,13 +234,41 @@ private boolean parseSource(HttpMessage message, Source source, int depth, Strin
             resourcesfound |= processAttributeElement(message, depth, baseURL, el, "src");
         }
 
+        // Process Table elements
+        elements = source.getAllElements(HTMLElementName.TABLE);
+        for (Element el : elements) {
+            resourcesfound |= processAttributeElement(message, depth, baseURL, el, "background");
+        }
+
+        // Process TD elements
+        elements = source.getAllElements(HTMLElementName.TD);
+        for (Element src : elements) {
+            resourcesfound |= processAttributeElement(message, depth, baseURL, src, "background");
+        }
+
+        // Process Video elements
+        elements = source.getAllElements(HTMLElementName.VIDEO);
+        for (Element el : elements) {
+            resourcesfound |= processAttributeElement(message, depth, baseURL, el, "src");
+            Source videoSource = new Source(el.getContent().toString());
```

Comment:
I think the earlier point here was that you should be able to `el.getAllElements(HTMLElementName.SOURCE)` without having to reparse into a new `Source` object.

http://jericho.htmlparser.net/docs/javadoc/net/htmlparser/jericho/Element.html (See "Methods inherited from class Segment")

Reply:
Thanks, really useful, amended :)

Reply:
Thanks for the quick tweak :grinning: